report/4.Design/RTS.tex

```diff
@@ -6,7 +6,74 @@ \subsection{Scheduling}\label{Design:Scheduling}
+Due to the fast execution time of the tasks, and that preemption and context switching would exceed the time for simply completing the previous task.
+At first it was at deemed obvious to use the cyclic executive model as all tasks should run regularly, and that they execute fast enough to run within the major cycle the cyclic execution method should be perfect. 
+However, should the execution time of a single task exceed one millisecond, the usage of cyclic executive is no longer viable, as the task to keep the USB connection alive is required to be executed once every millisecond.
+In this case, fixed-priority scheduling is a better alternative, as it allows the scheduler to pre-empt the task in order to run the USB related task.
 
-The cyclic executive method is a common way to implement hard real-time systems due to its simplicity, deterministic behavior and relative ease of implementation\cite{CyclicExecutionKimLarsen}.
\ No newline at end of file
+Both the fixed priority and cyclic executive methods are a common ways to implement hard real-time systems due to their simplicity, deterministic behavior and relative ease of implementation\cite{CyclicExecutionKimLarsen}.
+
+In order to ensure that the system is schedulable, an UPPAAL model of the scheduler and the tasks was created. 
+
+\subsubsection{UPPAAL}
+UPPAAL is an integrated tool environment for modeling, validation and verification of real-time systems modeled as networks of timed automata\cite{UPPAALWebsite}. 
+Using the UPPAAL tool, a model of the tasks of the system was constructed. 
+
+\paragraph{Templates}
+When creating an UPPAAL project, a templates must be made, which is a model that describes a system or a part of a system.  
+The templates are later used to make up the entire system, and they can communicate using channels specified in the declarations of the project.
+
+\paragraph{Locations}
+An UPPAAL model consists of at least one \textit{location}, which is a state that the system can be in. 
+A location can optionally be specified as \textit{initial}, \textit{urgent}, \textit{committed}. 
+If a location is marked as initial, it is defined as the starting state for the modelled system. 
+A location marked with urgent will be the preferred destination, if there are multiple possible transitions. 
+A location marked as committed is an location that is preferred over an location without any type and that is overruled by a location marked as urgent. 
+If a location is not marked with an specific type, the location might be taken but has the lowest priority if any other transitions are possible. 
+A location can be given an \textit{invariant} that can be used to specify a condition that must be fulfilled before a location can be left. 
+
+\paragraph{Edges}
+Between locations an edge can be created, they are used to model a shift in model state and is used to control the flow through the template. 
```

Comment:
```suggestion
Between locations are edges.
Edges are used to model a shift in model state and is used to control the flow through the template. 
```

Reply:
måske slå dem sammen

between locations are edges, which are used to mode....